### PR TITLE
IGNITE-15107 Integrate DDL support.

### PR DESCRIPTION
**Pull request by zstan**
https://issues.apache.org/jira/browse/IGNITE-15107

### CODE REVIEW

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ddl/DdlCommandHandler.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    }
+
+    /** Validate command. */
+    private void validateCommand(DdlCommand cmd) {
+        if (isNullOrEmpty(cmd.tableName()))
+            throw new IllegalArgumentException("tableName name is undefined.");
+    }
+
+    /** */
+    private void handle0(CreateTableCommand cmd) throws IgniteInternalCheckedException {
```
Comment (AMashenkov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ddl/DdlCommandHandler.java:85]: Let's rename to e.g. handleCreateCmd or smth meaningful.

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java [hunk 2]
```diff
@@ -273,30 +118,16 @@ public void ifNotExists(boolean ifNotExists) {
...
-        this.dataRegionName = dataRegionName;
-    }
-
-    /**
-     * @return Encrypted flag.
+     * @return Primary key columns.
      */
-    public boolean encrypted() {
-        return encrypted;
+    public List<String> primaryKeyColumns() {
```
Comment (AMashenkov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java:123]: Is it possible to make return types consistent in the next method's signatures?
`List<String> primaryKeyColumns` and `public Column[] columns() `

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java [hunk 3]
```diff
@@ -19,76 +19,33 @@
...
     private String schemaName;
 
-    /** Table name. */
-    private String tblName;
-
-    /** Cache name upon which new cache configuration for this table must be based. */
-    private String templateName;
-
     /** Name of new cache associated with this table. */
     private String cacheName;
```
Comment (AMashenkov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java:36]: Let's rename to tableName or even reuse from parent class?

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java [hunk 4]
```diff
@@ -19,76 +19,33 @@
...
 /**
  * CREATE TABLE statement.
  */
-public class CreateTableCommand implements DdlCommand {
+public class CreateTableCommand extends DdlCommand {
     /**
      * Schema name upon which this statement has been issued - <b>not</b> the name of the schema where this new table
      * will be created.
      */
     private String schemaName;
```
Comment (AMashenkov) [inline modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java:33]: Does it make sense to move schema to the parent class.

File: modules/schema/src/main/java/org/apache/ignite/internal/schema/configuration/SchemaConfigurationConverter.java [hunk 5]
```diff
@@ -589,7 +589,7 @@ else if (cls == Float.class)
         else if (cls == Double.class)
             return ColumnType.DOUBLE;
 
-        // Temporal types.
+        // Time interval types.
```
Comment (AMashenkov) [inline modules/schema/src/main/java/org/apache/ignite/internal/schema/configuration/SchemaConfigurationConverter.java:592]: We don't support interval types.
These are temporal, not interval.

File: modules/schema/src/main/java/org/apache/ignite/internal/schema/builder/TableColumnBuilderImpl.java [hunk 6]
```diff
@@ -71,7 +64,8 @@ public TableColumnBuilderImpl(String colName, ColumnType colType) {
     }
 
     /** {@inheritDoc} */
-    @Override public TableColumnBuilderImpl withHints(Map<String, String> hints) {
+    @Override public TableColumnBuilderImpl
```
Comment (korlov42) [inline modules/schema/src/main/java/org/apache/ignite/internal/schema/builder/TableColumnBuilderImpl.java:67]: line break

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java (hunk 2 above)
Comment (zstan) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/CreateTableCommand.java:123]: got it !

File: modules/schema/src/main/java/org/apache/ignite/internal/schema/configuration/SchemaConfigurationConverter.java (hunk 5 above)
Comment (zstan) [reply modules/schema/src/main/java/org/apache/ignite/internal/schema/configuration/SchemaConfigurationConverter.java:592]: change for: Date time types. hope it`s more clear.

File: modules/api/src/main/java/org/apache/ignite/sql/package-info.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Contains API classes for sql execution.
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/api/src/main/java/org/apache/ignite/sql/package-info.java:19]: change for : "Contains API classes for sql processing."

File: modules/api/src/main/java/org/apache/ignite/schema/SchemaTable.java [hunk 8]
```diff
@@ -75,4 +76,9 @@
      * @return Table modification builder.
      */
     TableModificationBuilder toBuilder();
+
+    /** @return Table with schema canonical name. */
+    static String canonicalName(@NotNull String schema, @NotNull String name) {
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/api/src/main/java/org/apache/ignite/schema/SchemaTable.java:81]: i move it into internal SchemaTableImpl#canonicalName

File: modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/DdlSqlToCommandConverter.java [hunk 9]
```diff
@@ -153,22 +140,33 @@ private CreateTableCommand convertCreateTable(IgniteSqlCreateTable createTblNode
...
             String name = col.name.getSimple();
-            RelDataType type = planner.convert(col.dataType);
+
+            RelDataType relType = planner.convert(col.dataType);
 
             Object dflt = null;
             if (col.expression != null)
                 dflt = ((SqlLiteral)col.expression).getValue();
 
-            cols.add(new ColumnDefinition(name, type, dflt));
```
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/ddl/DdlSqlToCommandConverter.java:171]: the comment is marked as resolved whereas neither changes nor reasons not to do so were provided

File: modules/calcite/src/main/codegen/includes/parserImpls.ftl [hunk 10]
```diff
@@ -43,27 +43,9 @@ SqlLiteral CreateTableOptionKey() :
 {
 }
 {
-    <TEMPLATE> { return SqlLiteral.createSymbol(IgniteSqlCreateTableOptionEnum.TEMPLATE, getPos()); }
-|
     <BACKUPS> { return SqlLiteral.createSymbol(IgniteSqlCreateTableOptionEnum.BACKUPS, getPos()); }
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/calcite/src/main/codegen/includes/parserImpls.ftl]: catch it ! i append both.

File: modules/api/src/main/java/org/apache/ignite/schema/builder/SchemaTableBuilder.java [hunk 11]
```diff
@@ -35,6 +35,14 @@
      */
     SchemaTableBuilder columns(Column... columns);
 
+    /**
+     * Adds columns to the table.
+     *
+     * @param columns Table columns.
+     * @return {@code This} for chaining.
+     */
+    SchemaTableBuilder columns(Iterable<Column> columns);
```
Comment (AMashenkov) [inline modules/api/src/main/java/org/apache/ignite/schema/builder/SchemaTableBuilder.java:44]: Let's drop `SchemaTableBuilder columns(Column... columns) `method,
as user always can call builder.columns(List.of(Columns[])

File: modules/api/src/main/java/org/apache/ignite/sql/IgniteSql.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+package org.apache.ignite.sql;
+
+import java.util.List;
+import org.apache.ignite.internal.util.Cursor;
+
+/**
+ * Interface that provides methods for sql queries.
+ */
+public interface IgniteSql {
```
Comment (AMashenkov) [inline modules/api/src/main/java/org/apache/ignite/sql/IgniteSql.java:26]:
```suggestion
public interface IgniteSQL {
```

**Review by AMashenkov — APPROVED** [review]

**Pull request closed without merging**